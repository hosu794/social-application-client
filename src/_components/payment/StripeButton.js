import React from "react";
import StripeCheckout from "react-stripe-checkout";
import axios from "axios";
import { useDispatch } from "react-redux";
import PropTypes from "prop-types";
import { paymentActions } from "../../_actions";

const StripeButton = ({ price }) => {
  const publishableKey =
    "pk_test_51HHQd4LYlgXeVHLv1FHC3ZDuVHoNlp8plCWpl7ATrvCATEr5TbjADjUg103jxLHN8Sd48E15SRsboSOdiylwo41W00MeR6h1uY";
  const stripePrice = price * 1000;

  const dispatch = useDispatch();

  const onToken = (token) => {
    console.log(token);
    alert("Success");
    dispatch(
      paymentActions.payPremium({
        amount: stripePrice,
        token,
      })
    );
  };

  return (
    <StripeCheckout
      amount={stripePrice}
      label="Buy premium"
      name="The Writer's Mind"
      image="https://simpleicon.com/wp-content/uploads/account.png"
      description={`Your total is ${price}$`}
      panelLabel="Buy premium"
      token={onToken}
      stripeKey={publishableKey}
      currency="USD"
    />
  );
};

StripeButton.propTypes = {
  price: PropTypes.string.isRequired,
};

export default StripeButton;
