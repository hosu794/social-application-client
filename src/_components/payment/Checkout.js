import React from "react";

import { Link } from "react-router-dom";
import StripeButton from "./StripeButton";

function Checkout() {
  return (
    <div>
      <div className="level-item has-text-centered">
        <div>
          <StripeButton price="10" />
        </div>
      </div>
    </div>
  );
}

export default Checkout;
