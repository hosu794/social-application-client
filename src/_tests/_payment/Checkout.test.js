import React from "react";

import configureStore from "redux-mock-store";
import { useDispatch, Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";

import Checkout from "../../_components/payment/Checkout";
import { mount } from "enzyme";
import { wrap } from "lodash";
import StripeButton from "../../_components/payment/StripeButton";

const mockStore = configureStore([]);

let store = mockStore({
  payment: {
    loading: false,
  },
});

describe("Test for the Checkout Component", () => {
  let TestingComponent = () => (
    <MemoryRouter>
      <Provider store={store}>
        <Checkout />
      </Provider>
    </MemoryRouter>
  );

  let wrapper = mount(<TestingComponent />);

  test("should render component correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should component be defined", () => {
    expect(Checkout).toBeDefined();
  });

  test("should exists StripeButton Component", () => {
    console.log(wrapper.debug());
    let element = wrapper.contains(<StripeButton price="10" />);
    expect(element).toEqual(true);
  });
});
