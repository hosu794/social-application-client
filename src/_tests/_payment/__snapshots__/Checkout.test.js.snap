// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for the Checkout Component should render component correctly 1`] = `ReactWrapper {}`;
